tests/test_postfix.py

```diff
@@ -0,0 +1,377 @@
+    """test relaying an email successfully to SMTP_RELAYHOST, using Postfix container"""
+
+    assert container_per_test.connection.run_expect(
+        [0], "nohup /usr/local/bin/mailhog > /dev/null 2>&1 &"
+    )
+
+    sendmail_cmd = 'echo "Subject: Test Email\n\nThis is a test email body." | sendmail -f user1@example.com user2@example.com'
+    assert container_per_test.connection.run_expect([0], sendmail_cmd)
+
+    assert "Mail queue is empty" in container_per_test.connection.check_output(
+        "sleep 10; mailq"
+    )
+
+    log = container_per_test.connection.file(
+        "/var/log/postfix.log"
+    ).content_string
+    for output in [
+        "from=<user1@example.com>",
+        "to=<user2@example.com>",
+        "relay=localhost[127.0.0.1]:1025",
+        "dsn=2.0.0",
+        "status=sent (250 Ok: queued as ",
+        "=@mailhog.example)",
+    ]:
+        assert output in log
+
+
+@pytest.mark.parametrize(
+    "container_per_test", POSTFIX_WITH_VIRTUAL_MBOX_ENABLED, indirect=True
+)
+def test_postfix_send_email_delivered_to_virtual_mbox(
+    container_per_test: ContainerData,
+) -> None:
+    """test sending an email successfully to virtual mailbox, using Postfix container"""
+
+    assert container_per_test.connection.run_expect(
+        [0], "chown -R vmail:vmail /var/spool/vmail/"
+    )
+    assert container_per_test.connection.run_expect(
+        [0], "nohup /usr/local/bin/mailhog > /dev/null 2>&1 &"
+    )
+
+    sendmail_cmd = 'echo "Subject: Test Email\n\nThis is a test email body." | sendmail -f user1@example.com user2@example.com'
+    assert container_per_test.connection.run_expect([0], sendmail_cmd)
+
+    assert "Mail queue is empty" in container_per_test.connection.check_output(
+        "mailq"
+    )
+
+    log = container_per_test.connection.file(
+        "/var/log/postfix.log"
+    ).content_string
+    for output in [
+        "from=<user1@example.com>",
+        "to=<user2@example.com>",
+        "relay=virtual",
+        "dsn=2.0.0",
+        "status=sent (delivered to maildir)",
+    ]:
+        assert output in log
+
+    assert container_per_test.connection.file(
+        "/var/spool/vmail/example.com/user2"
+    ).exists
+
+
+@pytest.mark.skipif(
+    OS_VERSION != "tumbleweed",
+    reason="openldap2 package only available on Tumbleweed (in the main system)",
+)
+@pytest.mark.parametrize(
+    "container_per_test", POSTFIX_WITH_LDAP_ENABLED, indirect=True
+)
+def test_postfix_with_ldap_and_email_delivered(
+    container_per_test: ContainerData,
+) -> None:
+    """test relaying an email successfully with ldap lookup, using Postfix container"""
+
+    assert container_per_test.connection.run_expect(
+        [0], "nohup /usr/local/bin/mailhog > /dev/null 2>&1 &"
+    )
+
+    assert container_per_test.connection.run_expect(
+        [0], "/entrypoint/openldap-entrypoint.sh /usr/sbin/slapd"
+    )
+    assert "0.0.0.0:389" in container_per_test.connection.check_output(
+        "ss -tunlp"
```

Comment:
It'd be nice if we could validate healthiness outside checking if the port is open. for example by doing a simple query to it.